### PR TITLE
Feature: Add migration step for Form Taxonomies

### PR DESCRIPTION
**Pull request by kjohnson**
Resolves [GIVE-1119]

## Description

This PR adds a new `FormTaxonomies` step to the form migration process.

## Affects

Form migrations.

## Testing Instructions

Create a v2 form with form tags, categories.
Migrating the form to v3 should carry over the terms.

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [x] Includes unit tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

[GIVE-1119]: https://stellarwp.atlassian.net/browse/GIVE-1119?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/FormMigration/Steps/FormTaxonomies.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Give\FormMigration\Steps;
+
+use Give\FormMigration\Contracts\FormMigrationStep;
+
+/**
+ * @unreleased
+ */
+class FormTaxonomies extends FormMigrationStep
```
Comment (jonwaldstein) [inline src/FormMigration/Steps/FormTaxonomies.php:10]: Should we add a `canHandle` method?  Potentially 

```taxonomy_exists('give_forms_tag') || taxonomy_exists('give_forms_category')```

**Review by jonwaldstein — APPROVED** [review]
@kjohnson looks good, thank you for the unit tests too!

Just had one question/suggestion, otherwise ready for QA 🌈

File: src/FormMigration/Steps/FormTaxonomies.php (hunk 1 above)
Comment (kjohnson) [reply src/FormMigration/Steps/FormTaxonomies.php:10]: The `canHandle` method would make sense if we split forms, categories unto separate tests.
But, this being in a single test this is a "and/or" situation.

**Comment by kjohnson** [comment]
Build for QA, see https://github.com/impress-org/givewp/actions/runs/10375261894.

**Review by rickalday — APPROVED** [review]
Passed manual QA tests.

**Pull request merged**